### ISSUE TITLE
Egg Hatching

### ISSUE THREAD

**Issue by staurakis** [jabbink/PokemonGoBot#1287]
**Description:**
I have seen that this has been discussed again, but although it is said that the order of the eggs hatched are :10km,5km, 2km, what i am experiencing is that always incubating the 2 and 5 first and after these and only if there are none of them to go for the 10kms which is quite problematic if i may say.....
I have checked it for many days now and it always behaves like that.
Is there a way (apart from the manual way) to possibly change this behaviour?

Leave this sentence in your issue as proof that you have read and used this template.

**Comment by FrancYescO** [comment]
actual behavious should be (develop branch not 0.6.0)
lowest km in infinite incubator
higghest km in the standard one

**Comment by staurakis** [comment]
Indeed i am running the develop branch (sorry for not mentioning it). And the thing is that I (as most of people) have only one incubator (the infinite one) and thus the behaviour is the one we both describe. 
What i am saying in that it should go the opposite way in the primary incubator since this is the best since the way it is right now, the only way to incubate 10km eggs is to have none 2 or 5 km eggs which is quite rare not to say impossible....

**Comment by FrancYescO** [comment]
i still prefer that the infinite incubator continue hatch the lowest km egg.

i can hatch (so throw away) 5x2km egg instead of a 10km one, and in the meanwhile i have more possibility that all these 2km eggs be replaced with a 10km/5km, so i'll start hatching a 10km egg only when i'm full of these.

**Comment by staurakis** [comment]
In fact, since the 10km eggs contain rarest pokemons you would first want to incubate these as more valuable eggs that produce more valuable pokemons. The way you describe, you will get stuck in a situation i am in, where i have 7 10km eggs and 2 of 2km and 5km that when the get hatched, they replaced (as more often found) by another 2 or 5. That way, the most 10 km eggs you can hatch consequently is 1 or 2 at most (since the odds are that they will be replaced again by 2 or 5).
So, you will everytime have from 7-8 10km eggs and doing practically nothing with them.....

**Comment by showei** [comment]
I changed it to always hatch 10km eggs first, if no 10km egg, hatch the lowest then.

**Comment by staurakis** [comment]
You mean on the current development branch? I presume i should download and build again right?

**Comment by showei** [comment]
no I didn't submit a pull request, it's very easy you can do it by yourself if you want, at HatchEggs.kt, replace 'freeIncubators.first().usesRemaining == 0'  with 'freeIncubators.first().usesRemaining == 0 && eggResult.eggKmWalkedTarget != (10).toDouble()'

**Comment by staurakis** [comment]
Thank you very much for sharing! I have changed it, and i am testing it!
Nevertheless, even like this, i strongly believe that the priorities on hatching should be 10,5,2km eggs with that order.

**Comment by showei** [comment]
10,2,5 is better, this way you have the biggest chance to get Snolax, this is the ultimate goal.

**Comment by staurakis** [comment]
oh! i didn't know that 10,2,5 is indeed the best way to have a Snorlax! If that's the case then indeed, that's the best order i agree.

**Comment by showei** [comment]
yeah, the actual best order is 2,5,10, but this way you probably got no 10km eggs hatched between lv25 to lv30. 10,2,5 make sure you hatch 10km egg asap and still refresh useless eggs fast enough.
